menas/src/main/scala/za/co/absa/enceladus/menas/services/OozieService.scala

```diff
@@ -22,13 +22,41 @@ import org.springframework.stereotype.Component
 
 import za.co.absa.enceladus.menas.models.OozieCoordinatorStatus
 import za.co.absa.enceladus.menas.repositories.OozieRepository
+import za.co.absa.enceladus.model.menas.scheduler.oozie.OozieSchedule
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 
 @Component
 class OozieService @Autowired() (oozieRepository: OozieRepository) {
 
+  import scala.concurrent.ExecutionContext.Implicits.global
+
   def isOozieEnabled: Boolean = oozieRepository.isOozieEnabled()
 
   def getCoordinatorStatus(coordinatorId: String): Future[OozieCoordinatorStatus] = {
     oozieRepository.getCoordinatorStatus(coordinatorId)
   }
+
+  def runNow(oozieSchedule: OozieSchedule, reportDate: Option[String]): Future[String] = {
+    val wfPath = oozieSchedule.activeInstance match {
+      case Some(instance) => instance.workflowPath
+      case None           => throw new IllegalArgumentException("Cannot run a non-active schedule.")
+    }
+    val reportDateString = reportDate match {
+      case Some(date) => date
```

Comment:
Question: What if date is of wrong format/pattern or generic nonsense. Isn't it an request incoming value?

Reply:
Oozie will still create a workflow job and that job will then fail as your spark-submit would fail after being unable to parse the `--report-date` arg. This will be then seen by users on the history server with an appropriate message. 
`Error: Match error in 'XXXX'. Option --report-date expects a date in 'yyyy-MM-dd' format`

Reply:
Ok, that's a non-damaging outcome. Still I would consider checking the input.
Might be a separate task of course.

Reply:
Btw, do we know if that format is hard-coded in Oozie or they use system Locale? 🤔

Reply:
This is parsed in Std/Conf jobs I believe. So the report date format is fixed.